meta/block.rb

```diff
@@ -204,3 +204,13 @@
   mutation 'foo(nil, &nil).baz {}'
   mutation 'foo(nil, &:bar__mutant__).baz {}'
 end
+
+Mutant::Meta::Example.add :block do
+  source 'loop { true }'
+
+  singleton_mutations
+  mutation 'true'
+  mutation 'loop { false }'
+  mutation 'loop { raise }'
```

Comment:
In theory this one is also quite redundant, as it always raises. its equivalent to `raise` in observable semantics.

The list of these cases is endless, its not a blocker. Just a thougt.

Ultimatively: If we had a symbolic execution engine, we could find "static execution path of known result" and filter these cases out statically without casing them.

This commit removes some (non harmful noise), which is a small win I like to see.

But ultimatively we'd add (never ending) implementation noise in filtering this noise out. I'm thinking about biting the bullet and making a symbolic execution engine for Ruby...

Reply:
Yeah. I would like to do some "lightweight" filtering of some of these obvious cases but we don't want to go too far in the weeds because there is no bottom here. I think this loop case is one of the better ones to filter because at least the `raise` case you've highlighted fails instantly and doesn't trigger a timeout. Before you have to enable coverage via timeouts and it takes as long as your timeout setting.

But yes, there is definitely a balance to strike here in terms of implementation complexity and how many mutations we are able to cull.